### PR TITLE
8317128: java/nio/file/Files/CopyAndMove.java failed with AccessDeniedException

### PR DESCRIPTION
**Pull request by bplb**
The change #15501 removed explicit throwing of a `FileAlreadyExistsException` in `copyToForeignTarget` for non-directories when the target exists and `REPLACE_EXISTING` is not specified, instead relying on `FileChannel.open` eventually to throw this exception. The test, however, reuses the same file path, and on Windows `CreateFile`, which is invoked by `open`, can throw an `AccessDeniedException` if the file exists and has been marked for deletion but not yet actually deleted. This change proposes to reinstate explicitly throwing a FAEE.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8317128](https://bugs.openjdk.org/browse/JDK-8317128): java/nio/file/Files/CopyAndMove.java failed with AccessDeniedException (**Bug** - P3)

### Reviewers
 * [Alan Bateman](https://openjdk.org/census#alanb) (@AlanBateman - **Reviewer**)
 * [Lance Andersen](https://openjdk.org/census#lancea) (@LanceAndersen - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/15991/head:pull/15991` \
`$ git checkout pull/15991`

Update a local copy of the PR: \
`$ git checkout pull/15991` \
`$ git pull https://git.openjdk.org/jdk.git pull/15991/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 15991`

View PR using the GUI difftool: \
`$ git pr show -t 15991`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/15991.diff">https://git.openjdk.org/jdk/pull/15991.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/15991#issuecomment-1741208655)

### CODE REVIEW

File: src/java.base/share/classes/java/nio/file/CopyMoveHelper.java [hunk 1]
```diff
@@ -137,10 +137,14 @@ static void copyToForeignTarget(Path source, Path target,
...
+        } else if (Files.exists(target))
+            throw new FileAlreadyExistsException(target.toString());
+
         // create directory or copy file
         if (sourceAttrs.isDirectory()) {
-            if (opts.replaceExisting)
-                Files.deleteIfExists(target);
             Files.createDirectory(target);
         } else {
             try (InputStream in = Files.newInputStream(source)) {
```
Comment (AlanBateman) [inline src/java.base/share/classes/java/nio/file/CopyMoveHelper.java:150]: If we are you reverting that then I assume you also should revert the change to L147 too, meaning replaceExistingOrEmpty() isn't needed now.

Comment (bplb) [reply src/java.base/share/classes/java/nio/file/CopyMoveHelper.java:142]: `replaceExistingOrEmpty()` removed in 4a34ff3b36bbcbd43139f7888ac4d953ecef91da.

**Review by AlanBateman — APPROVED** [review]

**Review by LanceAndersen — APPROVED** [review]
looks Ok to me

**Review by AlanBateman — APPROVED** [review]
